lib/Frontend/ModuleInterfaceLoader.cpp

```diff
@@ -1653,6 +1657,22 @@ InterfaceSubContextDelegateImpl::runInSubCompilerInstance(StringRef moduleName,
   if (subInvocation.parseArgs(SubArgs, *Diags)) {
     return std::make_error_code(std::errc::not_supported);
   }
+
+  // If the target triple parsed from the Swift interface file differs
+  // only in subarchitecture from the original target triple, then
+  // we have loaded a Swift interface from a different-but-compatible
+  // architecture slice. Use the original subarchitecture.
+  llvm::Triple parsedTargetTriple(subInvocation.getTargetTriple());
+  if (parsedTargetTriple.getSubArch() != originalTargetTriple.getSubArch() &&
+      parsedTargetTriple.getArch() == originalTargetTriple.getArch() &&
```

Comment:
Style thoughts, but I usually like to throw massive comparison-chains in a `std::tie`

e.g.

```c++
if (pTT.getSubArch() != oTT.getSubArch() &&
    std::tie(pTT.getArch(), pTT.getVendor(), pTT.getOS(), pTT.getEnvironment()) == std::tie(oTT.getArch(), oTT.getVendor(), oTT.getOS(), oTT.getEnvironment())) {
  //...
}
```
I feel like it hides the redundant `==` and shows what is equal and what is different more clearly.
I'll leave it up to you, but just throwing it out there.

Reply:
That's going to give us a really, really really long line (and also cause a merge conflict), so I think I'll leave this as-is.